### DISCUSSION TITLE
pinDependencies does not pin the dependencies in  Gemfile / Gemfile.lock subfolders

### DISCUSSION THREAD

**Discussion by petehl** [renovatebot/renovate#14126] [Request Help] [answered]
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### Describe the bug

We have a repsoitory containing ruby and js code in subfolders. 

We added `:pinDependencies` to our `config:base` as our dependencies were not updating. 
Renovate then opened up PR's to pin the dependencies in all the `package.json` and `yarn.lock` within the repo - and they now receive version updates. 🙌 

However the Gemfile(s) were not automatically updated, they still contained (~> 'x.x') and did not receive any updates.
I would expect that Renvoate also open PR's to pin the Gem dependencies. 

We've had to manually update the Gemfile to specific versions and regenerate the Gemfile.lock to get renovate to update the dependency versions in this projects. 

I've created a small test project here for experiementing with renovate config
https://github.com/petehl/renovate-test

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by HonkingGoose** [comment]
I'm not sure the reproduction is _minimal_ enough for our use. I think we prefer to have a reproduction more like this:

- One directory with one JavaScript dependency + `yarn.lock` that needs updating.
- And another directory with one Ruby dependency + lockfile that needs updating.

Also your `renovate.json` config contains a lot of rules, can you maybe try from a `config:base` preset, and try if the problem happens with that basic config as well?

**Comment by petehl** [comment]
I've reduced it to a more minimal repository now as requested along with removing a lot of the extra rules from the `renovate.json` 

As you can see, there has been a PR to pin the JS dependency, and an associated PR to upgrade after, however the Ruby gem is still behind.

**Comment by HonkingGoose** [comment]
Thank you for reducing it to a more minimal repository! ❤️

**Comment by StinkyLord** [comment]
Hi, Assign this to me please.

**Comment by StinkyLord** [comment]
![image](https://user-images.githubusercontent.com/56318487/152998527-fb59a60e-2efa-48b1-8891-4caf12c5e6ee.png)
adding dep.depType = "Dependencies"
solved the issue, 
apparently we are reading Ruby files and parsing them fine, but Dependency type wasn't assigned as "dependencies" so the versions weren't pinned for the gem dependency, ill open a PR

**Comment by rarkins** [comment]
@StinkyLord unfortunately I'm not sure we all understood the issue before. Bundler already sets `depTypes` instead of `depType`, so it's not correct to set `depType=dependencies`.

`:pinDependencies` has been around since Renovate supported only JS. So it meant "pin dependencies and not devDependencies". Unfortunately the description for this preset was not long enough. Here's what I think we need to do:
- Update the description for `:pinDependencies` to make clear it's referring to dependencies with depType=dependencies, as in `package.json`
- Retest this scenario using `rangeStrategy=pin` instead of `:pinDependencies` preset and see if it works as expected.

**Comment by HonkingGoose** [comment]
> * Update the description for `:pinDependencies` to make clear it's referring to dependencies with depType=dependencies, as in `package.json`

I think this is where we want to make the change then?

https://github.com/renovatebot/renovate/blob/64460a75f0aade72fc8f0c2fd8f034ff59ea549c/lib/config/presets/internal/default.ts#L85-L93

**Comment by StinkyLord** [comment]
to me :pinDependencies seemed like a global config for every dependency out there, the word Dependencies is global, and doesn't only imply that we are looking for the word `dependencies`
see
![image](https://user-images.githubusercontent.com/56318487/153222193-8c9c4a07-a76c-44eb-96ee-1c602dc24b19.png)

Changing the description could be a nice solution here, it should say at least what resolvers does it support.
cause if you go to Ruby Versioning, you can see that RangeStrategy Supports "Pin", 
but there's no connection what so ever to the fact that the `dependencies` from the `:pinDependencies` rule are referring to the Explicit word `dependencies` from `packages.json`

**Comment by StinkyLord** [comment]
@rarkins the `rangeStrategy=pin` works as expected, i haven't tried that, because i thought the `:pinDependencies` should do set it automatically for dependencies.
so this issue can be closed i believe, with a 
1) small documentation addition(what managers are supported using this flag)
2) description change in code can be an addition in my opinion

please advise.

**Comment by rarkins** [comment]
Let's change its behavior to set rangeStrategy=pin for all, but do so in a breaking release.

**Comment by StinkyLord** [comment]
Makes sense, although this task is full of chat and old description, maybe we should close it as wrong configuration used and relate it to another issue that we will open that describes the breaking change that we want to do.

I suggest that for now, just adding to the documentation temporary until that issue is resolved.

**Comment by rarkins** [comment, marked as answer]
Answer: `:pinDependencies` pins any dependencies with `depType=dependencies`, not "every single dependency you have". To pin all dependencies, set `rangeStrategy=pin`

**Reply by viceice** [reply]
We shoud update the description for those presets
